ddl/generated_column.go

```diff
@@ -137,6 +139,31 @@ func findColumnNamesInExpr(expr ast.ExprNode) []*ast.ColumnName {
 	return c.cols
 }
 
+// hasDependentByGeneratedColumn checks whether there are other columns depend on this column or not.
+func hasDependentByGeneratedColumn(tblInfo *model.TableInfo, colName model.CIStr) (bool, string) {
+	for _, col := range tblInfo.Columns {
+		for dep := range col.Dependences {
+			if dep == colName.L {
+				return true, dep
+			}
+		}
+	}
+	return false, ""
+}
+
+func isGeneratedRelatedColumn(tblInfo *model.TableInfo, newCol, col *model.ColumnInfo) error {
+	if newCol.IsGenerated() || col.IsGenerated() {
+		// TODO: Make it compatible with MySQL error.
+		msg := fmt.Sprintf("tidb_enable_change_column_type is true, newCol IsGenerated %v, oldCol IsGenerated %v", newCol.IsGenerated(), col.IsGenerated())
```

Comment:
`IsGenerated` is less friendly to read in the log.

Reply:
Any suggestions?

Reply:
it not big deal, i think